papermill_origami/noteable_dagstermill/assets.py

```diff
@@ -0,0 +1,315 @@
+import copy
+import os
+import sys
+import tempfile
+import uuid
+from base64 import b64encode
+from typing import Any, Dict, Mapping, Optional, Set, Union
+
+import cloudpickle as pickle
+import dagster._check as check
+import nbformat
+import papermill
+from dagster import (
+    AssetIn,
+    AssetKey,
+    MetadataValue,
+    Nothing,
+    Output,
+    PartitionsDefinition,
+    ResourceDefinition,
+    asset,
+)
+from dagster._core.definitions.events import CoercibleToAssetKeyPrefix
+from dagster._core.definitions.utils import validate_tags
+from dagster._core.execution.context.compute import SolidExecutionContext
+from dagster._core.execution.context.system import StepExecutionContext
+from dagster._utils import safe_tempfile_path
+from dagster._utils.error import serializable_error_info_from_exc_info
+from dagstermill import _load_input_parameter, _reconstitute_pipeline_context
+from dagstermill.compat import ExecutionError
+from dagstermill.factory import _find_first_tagged_cell_index, get_papermill_parameters
+from jupyter_client.utils import run_sync
+from origami.client import ClientConfig, NoteableClient
+from papermill.iorw import load_notebook_node, papermill_io, write_ipynb
+from papermill.translators import PythonTranslator
+
+from ..iorw import NoteableHandler
+from ..util import removeprefix
+from .context import SerializableExecutionContext
+
+
+def _dm_compute(
+    name,
+    notebook_path,
+    notebook_url,
+):
+    check.str_param(name, "name")
+    check.str_param(notebook_path, "notebook_path")
+    check.str_param(notebook_url, "notebook_url")
+
+    def _t_fn(context, **inputs):
+        check.inst_param(context, "context", SolidExecutionContext)
+        check.param_invariant(
+            isinstance(context.run_config, dict),
+            "context",
+            "StepExecutionContext must have valid run_config",
+        )
+
+        step_execution_context: StepExecutionContext = context.get_step_execution_context()
+
+        with tempfile.TemporaryDirectory() as output_notebook_dir:
+            with safe_tempfile_path() as output_log_path:
+
+                prefix = str(uuid.uuid4())
+                parameterized_notebook_path = os.path.join(
+                    output_notebook_dir, f"{prefix}-inter.ipynb"
+                )
+
+                executed_notebook_path = os.path.join(output_notebook_dir, f"{prefix}-out.ipynb")
+
+                # noteable specific client
+                noteable_client = NoteableClient(config=ClientConfig())
+                # We need to explicitly register the `NoteableHandler` with the client here,
+                # otherwise, `NoteableHandler.read` will try to create a new `NoteableClient`
+                # which will conflict with the one we have here.
+                papermill_io.register("noteable://", NoteableHandler(noteable_client))
+
+                # needs to be done before `load_notebook_node`
+                run_sync(noteable_client.__aenter__)()
```

Comment:
As Matt suggested, we can remove the explicit client creation here since origami pulls in the default env vars and creates the client if not passed in.

```suggestion
```